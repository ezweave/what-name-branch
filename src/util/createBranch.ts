import { execSync } from 'child_process';

interface ExecSync {
  (command: string): Buffer;
}

export interface CreateBranch {
  (execSyncFunction?: ExecSync): (name: string) => string;
}

export const createBranch: CreateBranch = (execSyncFunction = execSync) => (name) => {
  return execSyncFunction(`git checkout -b ${name}`).toString();
};
