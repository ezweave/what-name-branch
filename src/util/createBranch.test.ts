import { createBranch } from './createBranch';

describe(createBranch, () => {
  it('should create a branch with no problems', () => {
    const name = 'foo';
    const mockExec = jest.fn().mockImplementation(() => {
      return `Switched to a new branch '${name}'`;
    });
    expect(createBranch(mockExec)(name)).toMatchSnapshot();
  });
  it('should throw an error for an invalid branch name', async () => {
    const name = '...asdf';
    const mockExec = jest.fn().mockImplementation(() => {
      throw new Error(`fatal: '${name}' is not a valid branch name`);
    });
    expect(() => createBranch(mockExec)(name)).toThrowErrorMatchingSnapshot();
  });
});
