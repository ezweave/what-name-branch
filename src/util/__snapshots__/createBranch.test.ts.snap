// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBranch should create a branch with no problems 1`] = `"Switched to a new branch 'foo'"`;

exports[`createBranch should throw an error for an invalid branch name 1`] = `"fatal: '...asdf' is not a valid branch name"`;
